### PR TITLE
Update documentation

### LINKED ISSUE #12760: Update documentation for vector module
Corresponding to changes in #12720 and #12749, the documentation should be updated:
http://docs.sympy.org/dev/modules/vector/basics.html 
http://docs.sympy.org/dev/modules/vector/fields.html#scalar-and-vector-field-functionality

### PR DESCRIPTION
**Pull request by szymag**
Remove information about correlation between Del() and CoordSys3D.

I also update information about calculating directional derivative. It wasn't done after merging this PR: #12417

Closes #12760

### CODE REVIEW

File: sympy/vector/coordsysrect.py [hunk 1]
```diff
@@ -10,7 +10,16 @@
                                     SpaceOrienter, QuaternionOrienter)
 
 
-class CoordSysCartesian(Basic):
+def CoordSysCartesian(*args, **kwargs):
+    SymPyDeprecationWarning(
```
Comment (asmeurer) [inline sympy/vector/coordsysrect.py]: This needs a deprecation issue.

**Comment by asmeurer** [comment]
This is coming in very late. It might need to wait until after the first release candidate.

File: sympy/vector/coordsysrect.py (hunk 1 above)
Comment (szymag) [inline sympy/vector/coordsysrect.py]: I have already done it: https://github.com/sympy/sympy/issues/12865

**Comment by szymag** [comment]
> This is coming in very late. It might need to wait until after the first release candidate.

I didn't suspect that 1.1 version will be ready so fast.

File: sympy/vector/coordsysrect.py (hunk 1 above)
Comment (asmeurer) [inline sympy/vector/coordsysrect.py]: It needs to be specified as the `issue` argument. This is a required argument now in the `1.1` branch.

Comment (asmeurer) [inline sympy/vector/coordsysrect.py]: Also please add it to https://github.com/sympy/sympy/wiki/Release-Notes-for-1.1 (and add any other changes you know of for this release to that document).

**Comment by asmeurer** [comment]
> I didn't suspect that 1.1 version will be ready so fast.

Well we have a hard deadline, the codegen tutorial at the SciPy conference (a week from Monday). So the final version needs to go out sometime this next week.

**Comment by szymag** [comment]
Travis failed on this lines. I suppose that this test shouldn't throw exception. Moreover it didn't when these commits where in master.
```python
def test_coordsys3d():
    with warnings.catch_warnings():
        warnings.filterwarnings("ignore", category=SymPyDeprecationWarning)
        assert CoordSysCartesian("C") == CoordSys3D("C")
```

ping @moorepants

File: sympy/vector/coordsysrect.py (hunk 1 above)
Comment (szymag) [inline sympy/vector/coordsysrect.py]: > Also please add it to https://github.com/sympy/sympy/wiki/Release-Notes-for-1.1 (and add any other changes you know of for this release to that document).

I added information to release notes for every changes which I made in `vector` module.

> It needs to be specified as the issue argument. This is a required argument now in the 1.1 branch.

I don't understand. Could you explain what means "issue argument"?

Comment (szymag) [inline sympy/vector/coordsysrect.py]: > I don't understand. Could you explain what means "issue argument"?

Never mind. I saw how to do it in another PR.

**Comment by szymag** [comment]
Now it works, strange.

File: sympy/vector/deloperator.py [hunk 2]
```diff
@@ -16,7 +16,7 @@ def __new__(cls, system=None):
                 feature="delop operator inside coordinate system",
                 useinstead="it as instance Del class",
                 deprecated_since_version="1.1",
-                issue=12858,
```
Comment (asmeurer) [inline sympy/vector/deloperator.py]: There may have been some mixup here. This issue was supposed to say "vector" not "tensor". Since you already opened the other one with more information, should #12858 be closed?

Comment (asmeurer) [inline sympy/vector/deloperator.py]: If so, there is still a warning using this issue in operators.py.

Comment (szymag) [inline sympy/vector/deloperator.py]: > Since you already opened the other one with more information, should #12858 be closed? 

I did not information that you mentioned me in #12858  and I didn't notice this issue. It is the reason why I opened the other one. 
#12858 can be closed

Comment (szymag) [inline sympy/vector/deloperator.py]: > If so, there is still a warning using this issue in operators.py.

I opened proper issue #12884 and updated information in operators.py.
